File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {

        // for (auto v : points) {
        //     cout << v[0] << "," << v[1] << "  ";
        // }
        // cout << endl;

        sort(points.begin(), points.end(), [](auto a, auto b) {
            int dist1 = a[0] * a[0] + a[1] * a[1];
            int dist2 = b[0] * b[0] + b[1] * b[1];
            return dist2 > dist1;
        });

        // for (auto v : points) {
        //     cout << v[0] << "," << v[1] << "  ";
        // }
        // cout << endl;

        points.resize(k);
        return points;
    }
};

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.cs
public class Solution {
    public int[][] KClosest(int[][] points, int k) {
        var list = new List<int[]>(points);

        list.Sort((a, b) => {
            var dist1 = a[0]*a[0] + a[1]*a[1];
            var dist2 = b[0]*b[0] + b[1]*b[1];

            return dist1.CompareTo(dist2);
        });

        return list.Take(k).ToArray();
    }
}